jsapp/js/account/organization/OrganizationSettingsRoute.tsx

```diff
@@ -1,7 +1,173 @@
+  }
+
+  const mmoLabel = getSimpleMMOLabel(
+    envStore.data,
+    subscriptionStore.activeSubscriptions[0],
+    false,
+    true
+  );
+  const mmoLabelLowercase = mmoLabel.toLowerCase();
+
+  if (orgQuery.isLoading) {
+    return <LoadingSpinner />;
+  }
+
+  let deletionMessage = t('To delete this ##team/org##, please contact the server administrator.')
+    .replaceAll('##team/org##', mmoLabelLowercase);
+  if (isStripeEnabled) {
+    deletionMessage = t("To delete this ##team/org##, you need to cancel your current ##plan name## plan. At the end of the plan period your ##team/org##'s projects will be converted to projects owned by your personal account.")
+    .replaceAll('##team/org##', mmoLabelLowercase)
+    .replace('##plan name##', subscriptions.planName);
+  }
+
+  const currentTypeLabel = orgType === null ? '' : ORGANIZATION_TYPES[orgType]?.label;
+
   return (
-    <div>Organization settings view to be implemented</div>
+    <form className={styles.orgSettingsRoot} onSubmit={handleSave}>
+      <header className={styles.orgSettingsHeader}>
+        <h2 className={styles.orgSettingsHeaderText}>
+          {t('##team/org## details').replace('##team/org##', mmoLabel)}
+        </h2>
+      </header>
+
+      <section className={styles.fieldsRow}>
+        {/*
+          On all instances, both owner and admins should be able to edit
+          organization name.
+        */}
+        <TextBox
+          className={styles.field}
+          label={t('##team/org## name').replace('##team/org##', mmoLabel)}
+          value={name}
+          required
+          onChange={handleChangeName}
+          disabled={!isUserAdminOrOwner || orgQuery.isPending || patchOrganization.isPending}
+          errors={name === ''}
+        />
+
+        {/*
+          On Stripe-enabled instances, both owner and admins should be able to
+          edit organization website. On non-Stripe enabled instances it is not
+          visible.
+        */}
+        {isStripeEnabled && (
+          <TextBox
+            className={styles.field}
+            type='url'
+            label={t('##team/org## website').replace('##team/org##', mmoLabel)}
+            value={website}
+            required
+            onChange={handleChangeWebsite}
+            disabled={!isUserAdminOrOwner || orgQuery.isPending || patchOrganization.isPending}
+            errors={website === ''}
+          />
+        )}
+      </section>
+
+      {/*
+        On Stripe-enabled instances, both owner and admins should be able to
+        view organization type. On non-Stripe enabled instances it is not
+        visible.
+      */}
+      {isStripeEnabled && orgType && (
+        <section className={styles.fieldsRow}>
+          <KoboSelect
+            className={styles.fieldLong}
+            name='org-settings-type'
+            type='outline'
+            size='l'
+            isDisabled // always disabled
+            label={t('##team/org## type').replace('##team/org##', mmoLabel)}
```

Comment:
Non-blocking styling question - why runtime `replace()` when string templates are simpler?

```suggestion
            label={t(`${mmoLabel} type`)}
```

Reply:
> Non-blocking styling question - how it impacts translations using runtime `replace()` instead of simpler string templates?

I believe this is not styling. The translation function needs to be clear, no concatenations, so it can be identified for auto generation of translation strings. That's why we do a full sentence with a replace instead of a string template with code in it, which the `t` wouldn't recognize due to the dynamic string being passed to it.